src/components/lexical/plugins/ToolbarPlugin/components/LinkEditor.tsx

```diff
@@ -99,70 +124,83 @@ const LinkEditor = ({ isOpen, onOpen, onClose, insertLink }: LinkEditorProps) =>
   }, [editor, updateLinkEditor])
 
   const addLink = () => {
-    if (!lastSelection || !linkUrl) return
-    editor.dispatchCommand(TOGGLE_LINK_COMMAND, linkUrl)
+    if (!lastSelection) return
+    editor.dispatchCommand(
+      TOGGLE_LINK_COMMAND,
+      !!link ? ensureUrlProtocol(link) : null
+    )
     onClose()
   }
 
   return (
-    <Popover
-      initialFocusRef={initialFocusRef}
-      isOpen={isOpen}
-      onClose={onClose}
-      placement="top"
-    >
-      <PopoverTrigger>
-        <IconButton
-          onClick={shouldOpenEditor ? onOpen : insertLink}
-          isActive={isOpen}
-          aria-label="Insert Link"
-          icon={<Link />}
-        />
-      </PopoverTrigger>
-
-      <PopoverContent>
-        <PopoverHeader
-          px={2}
-          pt={2}
-          pb={0}
-          fontWeight="semibold"
-          fontSize="sm"
-          border="none"
-        >
-          Edit link
-        </PopoverHeader>
-
-        <PopoverArrow />
-        <PopoverCloseButton />
-
-        <PopoverBody p={2}>
-          <HStack>
-            <Input
-              ref={initialFocusRef}
-              size="sm"
-              placeholder="https://example.com"
-              value={linkUrl}
-              onChange={(event) => {
-                setLinkUrl(event.target.value)
-              }}
-              onKeyUp={(e) => {
-                if (e.key === "Enter") addLink()
+    <FormProvider {...methods}>
+      <Popover
+        initialFocusRef={initialFocusRef}
+        isOpen={isOpen}
+        onClose={onClose}
+        placement="top"
+      >
+        <PopoverTrigger>
+          <IconButton
+            onClick={shouldOpenEditor ? onOpen : insertLink}
+            isActive={isOpen}
+            aria-label="Insert Link"
+            icon={<Link />}
+          />
+        </PopoverTrigger>
+
+        <PopoverContent>
+          <PopoverHeader
+            px={2}
+            pt={2}
+            pb={0}
+            fontWeight="semibold"
+            fontSize="sm"
+            border="none"
+          >
+            Edit link
+          </PopoverHeader>
+
+          <PopoverArrow />
+          <PopoverCloseButton />
+          <PopoverBody p={2}>
+            <form
```

Comment:
You could use `methods.handleSubmit` on the button instead (the callback which you pass to this function will receive the form state as a param), in that case you wouldn't need to use `useWatch` ;)